build-release.sh

```diff
@@ -14,14 +14,64 @@ else
   echo "Disabling binary signing as config.sh does not define the required data."
 fi
 
-function sign {
+sign_windows() {
   if [ $can_sign == 0 ]; then
     return
   fi
   ./osslsigncode -pkcs12 ${SIGN_KEYSTORE} -pass "${SIGN_PASSWORD}" -n "${SIGN_NAME}" -i "${SIGN_URL}" -t http://timestamp.comodoca.com -in $1 -out $1-signed
   mv $1-signed $1
 }
 
+sign_macos() {
+  if [ ! -z "${OSX_HOST}" ]; then
+    osx_tmpdir=$(ssh "${OSX_HOST}" "mktemp -d")
+    reldir="$1"
+    binname="$2"
+    is_mono="$3"
+
+    if [[ "${is_mono}" == "1" ]]; then
+      appname="Godot_mono.app"
+      entitlements=editor_mono.entitlements
+      to_sign="Resources/GodotSharp/Tools/aot-compilers/iphone-arm64/aarch64-apple-darwin-mono-sgen MacOS/Godot"
+    else
+      appname="Godot.app"
+      entitlements=editor.entitlements
+      to_sign="MacOS/Godot"
+    fi
+
+    scp "${reldir}/${binname}.zip" "${OSX_HOST}:${osx_tmpdir}"
+    scp "${basedir}/build-macosx/${entitlements}" "${OSX_HOST}:${osx_tmpdir}"
+    ssh "${OSX_HOST}" "
+              cd ${osx_tmpdir} && \
+              unzip ${binname}.zip && \
+              for file in ${to_sign}; do \
+                echo \"Signing \$file:\" && \
+                codesign --deep --timestamp \
```

Comment:
@bruvzg I was lazy and used a `for` loop to sign both binaries so both use `--deep`, but it doesn't seem to be an issue.